joomla/joomla-cms
# [4.0] Added Save2Copy in Media Manager

Pull Request for Issue # .

### Summary of Changes

Added a button to the media manager edit page to save changes as a copy, without overwriting the original file.
![save2copy](https://user-images.githubusercontent.com/54251018/89663827-796cdd00-d8d6-11ea-951f-73020d623513.PNG)

The new files will be named like the original file with an added version
![new_name](https://user-images.githubusercontent.com/54251018/89663919-a1f4d700-d8d6-11ea-8661-a40f7bc16b4f.PNG)


### Testing Instructions

1. Navigate to content -> media
2. select an image and open the editor
3. make some changes and click 'Save as Copy'
4. make sure a copy of the file with the changes applied has been saved as 'original_filename-1.extension'
5. make sure the original file has not been modified
6. please verify that this also works in subfolders
7. [Change Plugin folder](https://github.com/joomla/joomla-cms/pull/30313#issuecomment-675368126)

### Actual result BEFORE applying this Pull Request

All changes have been applied to the original image in a destructive way. The original image could not be restored.

### Expected result AFTER applying this Pull Request

A copy of the original file is saved with changes applied, and the original file is not modified


### Documentation Changes Required
Added one new method `generateNewName` in 'administrator/components/com_media/src/Controller/ApiController.php' and documentation is added
